ipatests/test_xmlrpc/test_automember_plugin.py

```diff
@@ -220,13 +212,112 @@ def test_create_with_nonexistent_hostgroup(self, automember_hostgroup,
+                           match=r'%s: Automember rule not found' % grp.cn):
+            command()
+
+    @pytest.mark.parametrize(
+        "automember_grp, grp", [
+            ("automember_group", "group1"),
+            ("automember_hostgroup", "hostgroup1")
+        ])
+    def test_create_with_existent_group(self, automember_grp, grp, request):
+        """ Try to add a rule that already exists """
+        grp = request.getfixturevalue(grp)
+        automember_grp = request.getfixturevalue(automember_grp)
+        grp.ensure_exists()
+        automember_grp.ensure_exists()
+        command = automember_grp.make_create_command()
+        with pytest.raises(errors.DuplicateEntry,
+                           match=r'Automember rule with name "%s" '
+                                 r'already exists' % grp.cn):
+            command()
+
+
+class TestAutomemberFindNegative(XMLRPC_test):
+
+    @pytest.mark.parametrize(
+        "automember_grp, grp", [
+            ("automember_group", "group1"),
+            ("automember_hostgroup", "hostgroup1")
+        ])
+    def test_find_with_nonexistent_group(self, automember_grp, grp, request):
+        """ Try to find a rule with non-existent group/hostgroup """
+        grp = request.getfixturevalue(grp)
+        automember_grp = request.getfixturevalue(automember_grp)
+        grp.ensure_missing()
+        command = automember_grp.make_find_command()
+        result = command()
+        assert_deepequal(dict(
+            count=0,
+            truncated=False,
+            summary=u'0 rules matched',
+            result=[],
+        ), result)
+
+    @pytest.mark.parametrize(
+        "automember_grp, grp", [
+            ("automember_group", "group1"),
+            ("automember_hostgroup", "hostgroup1")
+        ])
+    def test_find_with_invalidtype(self, automember_grp, grp, request):
+        """ Try to find group/hostgroup rule with invalid type """
+        grp = request.getfixturevalue(grp)
+        automember_grp = request.getfixturevalue(automember_grp)
+        grp.ensure_exists()
+        automember_grp.ensure_exists()
+        command = automember_grp.make_command('automember_find', grp.name,
+                                              type='badtype')
+        with pytest.raises(errors.ValidationError,
+                           match=r'invalid \'type\': must be one '
+                                 r'of \'group\', \'hostgroup\''):
+            command()
+
+
+class TestAutomemberShowNegative(XMLRPC_test):
+
+    @pytest.mark.parametrize(
+        "automember_grp, grp", [
+            ("automember_group", "group1"),
+            ("automember_hostgroup", "hostgroup1")
+        ])
+    def test_show_with_nonexistent_group(self, automember_grp, grp, request):
+        """ Try to show a rule with non-existent group and hostgroup """
+        grp = request.getfixturevalue(grp)
+        automember_grp = request.getfixturevalue(automember_grp)
+        grp.ensure_missing()
```

Comment:
It should be automember_grp.ensure_missing().
The function can be simplified and use only automember_grp + request parameters (grp is not useful here, grp.cn can be replaced with automember_grp.cn).